src/aof.c

```diff
@@ -1060,6 +1061,19 @@ void feedAppendOnlyFile(int dictid, robj **argv, int argc) {
  * AOF loading
  * ------------------------------------------------------------------------- */
 
+/* Mark that we are loading INCR AOFs in the global state and setup the fields
+ * needed to provide loading stats. */
+void startLoadingIncrAofFiles(size_t size) {
+    startLoading(size, RDBFLAGS_AOF_PREAMBLE, 0);
+}
+
+/* Refresh the AOF loading progress info */
+void loadingIncrAofProgress(off_t size) {
+    server.loading_loaded_bytes += size;
+    if (server.stat_peak_memory < zmalloc_used_memory())
+        server.stat_peak_memory = zmalloc_used_memory();
+}
```

Comment:
maybe move that to be next to loadingProgress / startLoadingFile, and rename to just `loadingIncrProgress`.
i.e. a generic progress report that reports the delta rather than absolute.

Reply:
Modified